pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -389,6 +366,25 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 				}
 			}
 		}
+
+		// check all hostname in vHostDedupMap and if is not exist with HttpsRedirect set to true
+		// create VirtualHost to redirect
+		for _, hostname := range server.Hosts {
+			if vHost, exists := vHostDedupMap[host.Name(hostname)]; exists {
+				if server.Tls != nil && server.Tls.HttpsRedirect {
+					vHost.RequireTls = route.VirtualHost_ALL
+				}
+				continue
+			} else if server.Tls != nil && server.Tls.HttpsRedirect {
+				newVHost := &route.VirtualHost{
+					Name:                       domainName(hostname, port),
+					Domains:                    buildGatewayVirtualHostDomains(hostname, port),
+					IncludeRequestAttemptCount: true,
+					RequireTls:                 route.VirtualHost_ALL,
+				}
+				vHostDedupMap[host.Name(hostname)] = newVHost
+			}
+		}
```

Comment:
```suggestion
		for _, hostname := range server.Hosts {
			if !server.GetTls().GetHttpsRedirect() {
				continue
			}
			if vHost, exists := vHostDedupMap[host.Name(hostname)]; exists {
				vHost.RequireTls = route.VirtualHost_ALL
				continue
			}
			newVHost := &route.VirtualHost{
				Name:                       domainName(hostname, port),
				Domains:                    buildGatewayVirtualHostDomains(hostname, port),
				IncludeRequestAttemptCount: true,
				RequireTls:                 route.VirtualHost_ALL,
			}
			vHostDedupMap[host.Name(hostname)] = newVHost
		}
	}
```

Feels a little cleaner than duplicating the check multiple times

Reply:
Not required though, just personal preference

Reply:
yeah seems cleaner, let me try to commit the changes